File: src/test/resources/oracle/commits/rm-action-test-1600b87d81dac9889f5603e07849e6bc89995332/kotlin/OrderProcessor.kt
class OrderProcessor {
    private val taxRate = 0.08

    fun buildReceipt(customerName: String, unitPrice: Int, quantity: Int, discount: Int): String {
        val subtotal = unitPrice * quantity
        val discountedSubtotal = subtotal - discount
        val tax = (discountedSubtotal * taxRate).toInt()
        val total = discountedSubtotal + tax

        val header = formatHeader(customerName)
        val dollars = centsToDollars(total)
        val status = formatPaymentStatus(total, discount)

        return header + "\n" +
                "Subtotal: " + subtotal + "\n" +
                "Discount: " + discount + "\n" +
                "Tax: " + tax + "\n" +
                "Total: $" + dollars + "\n" +
                status
    }

    private fun formatHeader(customerName: String): String {
        return "Receipt for " + customerName.trim().uppercase()
    }

    private fun centsToDollars(cents: Int): Double {
        return cents / 100.0
    }

    private fun formatPaymentStatus(total: Int, discount: Int): String {
        if (total > 0 && discount > 0) {
            return "Discount applied"
        }
        if (total > 0) {
            return "Payment required"
        }
        return "No payment required"
    }
}

File: src/test/resources/oracle/commits/rm-action-test-581cc443bce22dc82f075cf6815c067a2b57429f/javascript/OrderProcessor.js
export class OrderProcessor {
    constructor() {
        this._taxRate = 0.08;
    }

    buildReceipt(customerName, unitPrice, quantity, discount) {
        const subtotal = unitPrice * quantity;
        const discountedSubtotal = subtotal - discount;
        const tax = Math.floor(discountedSubtotal * this._taxRate);
        const total = discountedSubtotal + tax;

        const header = this.formatHeader(customerName);
        const dollars = this.centsToDollars(total);
        const status = this.formatPaymentStatus(total, discount);

        return (
            `${header}\n` +
            `Subtotal: ${subtotal}\n` +
            `Discount: ${discount}\n` +
            `Tax: ${tax}\n` +
            `Total: $${dollars}\n` +
            `${status}`
        );
    }

    formatHeader(customerName) {
        return "Receipt for " + customerName.trim().toUpperCase();
    }

    centsToDollars(cents) {
        return cents / 100.0;
    }

    formatPaymentStatus(total, discount) {
        if (total > 0 && discount > 0) {
            return "Discount applied";
        }
        if (total > 0) {
            return "Payment required";
        }
        return "No payment required";
    }
}
